packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdInSummary.spec.ts

```diff
@@ -135,10 +151,132 @@ describeCompat("Create data store with group id", "NoCompat", (getTestObjectProv
+		assert(runtime3.storage.getSnapshot !== undefined, "getSnapshot should be defined");
+		const snapshot = await runtime3.storage.getSnapshot();
+		const channelsTree = snapshot.snapshotTree.trees[".channels"];
+		const mainObjectTree = channelsTree.trees[mainObject.id];
+		const dataObjectATree = channelsTree.trees[dataObjectA.id];
+		const dataObjectBTree = channelsTree.trees[dataObjectB.id];
+
+		assert(mainObjectTree.omitted === undefined, "mainObject should not be omitted");
+		assert(mainObjectTree.groupId === undefined, "mainObject should not have a groupId");
+		assert(Object.entries(mainObjectTree.trees).length > 0, "mainObject missing trees");
+		assert(Object.entries(mainObjectTree.blobs).length > 0, "mainObject missing blobs");
+
+		assert(dataObjectATree.omitted, "dataObjectA should be omitted");
+		assert(dataObjectATree.groupId === loadingGroupId, "dataObjectA should have a groupId");
+		assert(Object.entries(dataObjectATree.trees).length === 0, "dataObjectA has trees!");
+		assert(Object.entries(dataObjectATree.blobs).length === 0, "dataObjectA has blobs!");
+
+		assert(dataObjectBTree.omitted, "dataObjectB should be omitted");
+		assert(dataObjectBTree.groupId === loadingGroupId, "dataObjectB should have a groupId");
+		assert(Object.entries(dataObjectBTree.trees).length === 0, "dataObjectB has trees!");
+		assert(Object.entries(dataObjectBTree.blobs).length === 0, "dataObjectB has blobs!");
+
+		// Testing the get snapshot call with loadingGroupId
+		const loadingGroupIdSnapshot = await runtime3.storage.getSnapshot({
+			loadingGroupIds: [loadingGroupId],
+			versionId: summaryVersion,
+		});
+		const channelsTree2 = loadingGroupIdSnapshot.snapshotTree.trees[".channels"];
+		const mainObjectTree2 = channelsTree2.trees[mainObject.id];
+		const dataObjectATree2 = channelsTree2.trees[dataObjectA.id];
+		const dataObjectBTree2 = channelsTree2.trees[dataObjectB.id];
+
+		assert(mainObjectTree2.omitted, "mainObject should be omitted");
+		assert(mainObjectTree2.groupId === undefined, "mainObject should not have a groupId");
+		assert(Object.entries(mainObjectTree2.trees).length === 0, "mainObject has trees!");
+		assert(Object.entries(mainObjectTree2.blobs).length === 0, "mainObject has blobs!");
+
+		assert(dataObjectATree2.omitted === undefined, "dataObjectA should not be omitted");
+		assert(dataObjectATree2.groupId === loadingGroupId, "dataObjectA should have a groupId");
+		assert(Object.entries(dataObjectATree2.trees).length > 0, "dataObjectA missing trees");
+		assert(Object.entries(dataObjectATree2.blobs).length > 0, "dataObjectA missing blobs");
+
+		assert(dataObjectBTree2.omitted === undefined, "dataObjectB should not be omitted");
+		assert(dataObjectBTree2.groupId === loadingGroupId, "dataObjectB should have a groupId");
+		assert(Object.entries(dataObjectBTree2.trees).length > 0, "dataObjectB missing trees");
+		assert(Object.entries(dataObjectBTree2.blobs).length > 0, "dataObjectB missing blobs");
+	});
 });
```

Comment:
And one more: How do we test new runtime / old service, and old runtime / new service?
The former is more interesting IMHO (as I'd expect the latter to be more of responsibility of a service to test).

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/7312